internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -131,6 +141,57 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 	return ctrl.Result{}, nil
 }
 
+// unstructuredNode returns a raw unstructured from a Node.
+func unstructuredNode(node *corev1.Node) *unstructured.Unstructured {
+	obj := &unstructured.Unstructured{}
+	obj.SetAPIVersion(node.String())
+	obj.SetKind(node.Kind)
```

Comment:
node.String returns something different, we should use: obj.SetAPIVersion(node.GroupVersionKind().GroupVersion().String())

But I'm not sure if apiVersion and kind are actually set here. Afaik usually they are not set by client-go / CR (?)

The following should work 100%
```go
	obj.SetAPIVersion(corev1.SchemeGroupVersion.String())
	obj.SetKind("Node")
```